Hello, I am failing to fix the linux_64 build for may C++ and Python binding project. I tried to reduce CMAKE_BUILD_PARALLEL_LEVEL, tried to force Ninja compilation instead of make, attempted to modify my pyproject.toml in various ways. The error message in the pipeline is quite cryptic and does not provide enough explanation.

I attempted to set up a docker image similar to your runner on my local machine. The Linux build failed first time because of some internal memory issue:

 ```
 [65/165] /usr/bin/g++  -pthread -B /root/miniforge3/envs/snt/compiler_compat -DCODE_VERSION=\"0.5.2\" -I/scinumtools3/include -I/scinumtools3/src -O3 -DNDEBUG -std=gnu++20 -fPIC -Wno-deprecated -fmax-errors=3 -MD -MT src/snt/puq/CMakeFiles/snt-puq.dir/quantity.cpp.o -MF src/snt/puq/CMakeFiles/snt-puq.dir/quantity.cpp.o.d -o src/snt/puq/CMakeFiles/snt-puq.dir/quantity.cpp.o -c /scinumtools3/src/snt/puq/quantity.cpp
  ninja: build stopped: subcommand failed.


  *** CMake build failed
  error: subprocess-exited-with-error

  × Building wheel for scinumtools3 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> No available output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /root/miniforge3/envs/snt/bin/python /root/miniforge3/envs/snt/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /tmp/tmpeehwu8hv
  cwd: /scinumtools3
  Building wheel for scinumtools3 (pyproject.toml) ... error
  ERROR: Failed building wheel for scinumtools3
Failed to build scinumtools3
error: failed-wheel-build-for-install

× Failed to build installable wheels for some pyproject.toml based projects
╰─> scinumtools3
Exception information:
Traceback (most recent call last):
  File "/root/miniforge3/envs/snt/lib/python3.12/site-packages/pip/_internal/cli/base_command.py", line 109, in _run_wrapper
    status = _inner_run()
             ^^^^^^^^^^^^
  File "/root/miniforge3/envs/snt/lib/python3.12/site-packages/pip/_internal/cli/base_command.py", line 102, in _inner_run
    return self.run(options, args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniforge3/envs/snt/lib/python3.12/site-packages/pip/_internal/cli/req_command.py", line 101, in wrapper
    return func(self, options, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/miniforge3/envs/snt/lib/python3.12/site-packages/pip/_internal/commands/install.py", line 531, in run
    raise InstallWheelBuildError(build_failures)
pip._internal.exceptions.InstallWheelBuildError: <InstallWheelBuildError: failed-wheel-build-for-install>
Removed build tracker: '/tmp/pip-build-tracker-umx9qdzs'
```

but it succeeded when I lowered the CMAKE_BUILD_PARALLEL_LEVEL.
Nevertheless, the Linux build on your runner fails again and again.
Can you please tell me where I should look for more clues?

--- a/recipes/scinumtools3/meta.yaml
+++ b/recipes/scinumtools3/meta.yaml
@@ -1,5 +1,5 @@
 {% set name = "scinumtools3" %}
-{% set version = "0.5.4" %}
+{% set version = "0.5.5" %}
 
 package:
   name: {{ name }}
@@ -7,7 +7,7 @@ package:
 
 source:
   url: https://github.com/vrtulka23/scinumtools3/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: 10dd43da7e67a5f7f12b0c0cc0130a6a1e9103e601c9763989386df6484af714
+  sha256: dd66b696eedd719b8c4bb02dd0635c00ac2827fd75dbc0a326fcf842ab282f69
 
 build:
   number: 0

--- a/recipes/scinumtools3/meta.yaml
+++ b/recipes/scinumtools3/meta.yaml
@@ -13,6 +13,7 @@ build:
   number: 0
   script: 
     - export CMAKE_BUILD_PARALLEL_LEVEL=1  # [unix]
+    - export CMAKE_GENERATOR=Ninja         # [unix]
     - {{ PYTHON }} -m pip install . --no-deps -vv
 
 requirements:

--- a/recipes/scinumtools3/meta.yaml
+++ b/recipes/scinumtools3/meta.yaml
@@ -1,5 +1,5 @@
 {% set name = "scinumtools3" %}
-{% set version = "0.5.3" %}
+{% set version = "0.5.4" %}
 
 package:
   name: {{ name }}
@@ -7,7 +7,7 @@ package:
 
 source:
   url: https://github.com/vrtulka23/scinumtools3/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: 7e0e0d0d5c8328a4686362b489b4e70ac757b635cb783b474dce71aaaf937ac0
+  sha256: 10dd43da7e67a5f7f12b0c0cc0130a6a1e9103e601c9763989386df6484af714
 
 build:
   number: 0

